Issue 9 address pull request

JuanMiguelRoman1008 opened the pull request.

Commit:
Update res status to 201

Commit:
Linked services to search routes

Commit:
Restructure search routes to use database controllers

Commit:
Restructure Delete Route to use service controllers

Commit:
Restructure join request

Commit:
Update leave request route to use database controllers

Commit:
Update route descriptions

Commit:
Separate route and from database controllers

Commit:
Destructure req

Commit:
Destructure req in join route

Commit:
Destructure req in unjoin route

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+        return;
+    } catch (error) {
+        if (error.message === "NOT_FOUND") {
+            res.status(404).send(`Search ID does not exist`);
```
If you're not going to interpolate values into the string, you can just use the usual string form (e.g. `"Search ID does not exist") instead of using template literals (the backtick strings).

This is a sample of string interpolation:
```js
res.status(500).send(`An error occured while trying to create your search, ${error}`);
```

JuanMiguelRoman1008 replied:
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+        return;
+    } catch (error) {
+        if (error.message === "NOT_FOUND") {
+            res.status(404).send(`Search ID does not exist`);
```
Please don't use template literals if you're not going to interpolate values. Use regular strings instead.

JuanMiguelRoman1008 replied:
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+            res.status(404).send(`Search ID does not exist`);
+            return;
+        } else if (error.message === "ALREADY_EXISTS") {
+            res.status(409).send(`Player ID is already in Search`);
```
Please don't use template literals if you're not going to interpolate values. Use regular strings instead.

JuanMiguelRoman1008 replied:
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+            return;
+        } else {
+            res.status(500).send(
+                `An error occured while trying to join the search`
```
Please don't use template literals if you're not going to interpolate values. Use regular strings instead.

JuanMiguelRoman1008 replied:
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+            return;
+        } else {
+            res.status(500).send(
+                `An error occured while trying to join the search`
```
Did you intend to not add the error message for this unexpected generic error? (e.g. `error` or `error.message`)?

JuanMiguelRoman1008 replied:
I have added the error body when returning status 500
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+        return;
+    } catch (error) {
+        if (error.message === "NOT_FOUND") {
+            res.status(404).send(`Search ID does not exist`);
```
Please don't use template literals if you're not going to interpolate values. Use regular strings instead.

JuanMiguelRoman1008 replied:
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+            res.status(404).send(`Search ID does not exist`);
+            return;
+        } else if (error.message === "PLAYER_NOT_FOUND") {
+            res.status(404).send(`Player ID is not in Search`);
```
Did you intend to not add the error message for this unexpected generic error? (e.g. `error` or `error.message`)?

JuanMiguelRoman1008 replied:
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+            return;
+        } else {
+            res.status(500).send(
+                `An error occured while trying to join the search`
```
Please don't use template literals if you're not going to interpolate values. Use regular strings instead.

JuanMiguelRoman1008 replied:
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on routes/api/search.js:
```diff
@@ -1,36 +1,95 @@
+            return;
+        } else {
+            res.status(500).send(
+                `An error occured while trying to join the search`
```
Did you intend to not add the error message for this unexpected generic error? (e.g. error or error.message)?

JuanMiguelRoman1008 replied:
Addressed in d731ad8f2fcf384f5a3a3ae29c1523492965265b

jonfelixrico commented on services/search-repository.js:
```diff
@@ -0,0 +1,66 @@
+
+    search.party.splice(
+        search.party.findIndex((player) => {
+            player.user === playerID;
```
Can you verify if this works properly?

The callback `(player) => { player.user === playerID }` will always return `undefined` because you're not returning a value in the method body. The splice will not execute since the value returned by `findIndex` will always be `-1`.

For it to work, it should be either:

```js
(player) => {
  // notice the return?
  return player.user === playerID;
}
```

or

```js
(player) => player.user === playerID
// this is equivalent to the example above
```

To see the similarities/differences between the two examples, please refer to the [syntax documentation for arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#syntax).

JuanMiguelRoman1008 replied:
Addressed in a8aba9d2e6abbf19ec91ab9dea2919e2adba3c9c

jonfelixrico replied:
Tip: you can just use `(player) => player.user === playerID` instead of the longer form of

```js
(player) => {
  return player.user === playerID;
}
```

The single-statement style (the first one) should be maximized for simple statements.

jonfelixrico commented on services/search-repository.js:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    search.party.splice(
+        search.party.findIndex((player) => {
```
It would be more readable if you initialize a variable for `findIndex` on top of splice.

Actually you can improve this block of code and the if above it by:

```
const indexInParty = find index code
if (indexInParty === -1) {
  throw new Error("PLAYER_NOT_FOUND");
}

search.party.splice(indexInParty, 1)
```

`findIndex` can act like `some`. The difference is that if an element exists in the array, `some` will return `true` but `findIndex` will return a non-negative integer which represents the index of the first matching item found in the array.

If an element does not exist, then `some` will return `false` and `findIndex` will return `-1` to signify that the item is not found. See [the return value documentation for `findIndex` for more info](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex).

JuanMiguelRoman1008 replied:
Addressed in af51925f58fcd0345406ae68420d232d38f51319

jonfelixrico reviewed (changes requested):
First pass done for this PR.

jonfelixrico commented on services/search-repository.js:
```diff
@@ -0,0 +1,66 @@
+const Search = require("../models/Search");
+
+addSearch = async ({ user, game, limit }) => {
```
Please add `const` before the variable name since this is an initialization (e.g. `const myVariableName = async () => {}`).

You can also opt to just go with the traditional function declaration, e.g.:
```javascript
function myFunctionName () {
}
```

Using either in this use case should be ok since you're not using `this` inside your method.

jonfelixrico replied:
Please apply this to other methods in this file as well.

JuanMiguelRoman1008 replied:
Addressed in fdb073c1ed2d68842125314cf82163988b97a481

Commit:
Add request parameters in error message

Commit:
Add return statement for splice function

Commit:
Optimize execution and readability of leaveSearch function

Commit:
Add const to search-repository functions

JuanMiguelRoman1008 commented:
Done addressing the concerns first pass of code review
![image](https://user-images.githubusercontent.com/25260027/170054358-9baec476-1cad-4ad0-9c94-ece60fedb74d.png)
